src/widgets/DashboardPage.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import globalStyles from '../globalStyles';
+import { testData } from '../database/utilities/ReportDemoData';
+import { ReportSideBar } from './ReportSideBar';
+import { ReportChart } from './ReportChart';
+import { createOrUpdateRecord } from '../sync/incomingSyncUtils';
+
+export const DashboardPage = ({ database, settings }) => {
+  const [reports, setReports] = useState(null);
+  const [selectedItemIndex, setSelectedItemIndex] = useState(0);
+
+  // Used for adding data within ./ReportDemoData to the database for
+  // testing while syncing from mobile is not functional.
+  database.write(() =>
+    testData.forEach(report => {
+      const recordCopy = { ...report };
+      recordCopy.data = JSON.stringify(report.data);
+      createOrUpdateRecord(database, settings, 'Report', recordCopy);
+    })
+  );
+
+  useEffect(() => {
+    setReports(
+      database
+        .objects('Report')
+        .snapshot()
+        .then((report, index) => ({
+          id: report.reportID,
+          index,
+          title: report.title,
+          type: report.type,
+          date: report.date,
+          data: JSON.parse(report.data),
+        }))
+    );
+  }, [reports]);
```

Comment:
in the `useEffect`, we `setReports`, changing the `reports` variable. Then, we run the `useEffect` when `reports` changes. See a problem with this? :)